Sources/MapboxMaps/Foundation/Extensions/Core/DownloadStatus.swift

```diff
@@ -0,0 +1,122 @@
+@_exported import MapboxCommon
+@_implementationOnly import MapboxCommon_Private
+
+extension DownloadError: LocalizedError {
+    /// Standardized error message
+    public var errorDescription: String? {
+        return "\(code): \(message)"
+    }
+}
+
+extension DownloadStatus {
```

Comment:
Could this concept be modeled more cleanly using an enum with associated values? I'd love to avoid needing to lean so heavily on documentation to say in which states things will be nil or not.